src/base/bittorrent/session.cpp

```diff
@@ -4351,6 +4262,39 @@ void Session::handleExternalIPAlert(libt::external_ip_alert *p)
     Logger::instance()->addMessage(tr("External IP: %1", "e.g. External IP: 192.168.0.1").arg(p->external_address.to_string(ec).c_str()), Log::INFO);
 }
 
+void Session::handleStartUpFinished()
+{
+    m_startupInProgress = false;
```

Comment:
This flag is confusing. It splits lifetime of Session in 3 steps (uninitialized, starting, started), but we can handle only last two.
It's better to use m_started flag (initially false and assigning true here).

Reply:
OK. That seems to work too.